pkgs/test_core/lib/src/runner/reporter/github.dart

```diff
@@ -0,0 +1,196 @@
+  }) =>
+      GithubReporter._(engine, sink, printPath);
+
+  GithubReporter._(this._engine, this._sink, this._printPath) {
+    _subscriptions.add(_engine.onTestStarted.listen(_onTestStarted));
+    _subscriptions.add(_engine.success.asStream().listen(_onDone));
+  }
+
+  @override
+  void pause() {
+    if (_paused) return;
+    _paused = true;
+
+    for (var subscription in _subscriptions) {
+      subscription.pause();
+    }
+  }
+
+  @override
+  void resume() {
+    if (!_paused) return;
+    _paused = false;
+
+    for (var subscription in _subscriptions) {
+      subscription.resume();
+    }
+  }
+
+  void _cancel() {
+    for (var subscription in _subscriptions) {
+      subscription.cancel();
+    }
+    _subscriptions.clear();
+  }
+
+  /// A callback called when the engine begins running [liveTest].
+  void _onTestStarted(LiveTest liveTest) {
+    // Convert the future to a stream so that the subscription can be paused or
+    // canceled.
+    _subscriptions.add(
+        liveTest.onComplete.asStream().listen((_) => _onComplete(liveTest)));
+
+    // Collect messages from tests as they are emitted.
+    _subscriptions.add(liveTest.onMessage.listen((message) {
+      _testMessages.putIfAbsent(liveTest, () => []).add(message);
+    }));
+  }
+
+  /// A callback called when [liveTest] finishes running.
+  void _onComplete(LiveTest test) {
+    final errors = test.errors;
+    final messages = _testMessages[test] ?? [];
+    final skipped = test.state.result == Result.skipped;
+    final failed = errors.isNotEmpty;
+
+    void emitMessages(List<Message> messages) {
+      for (var message in messages) {
+        _sink.writeln(message.text);
+      }
+    }
+
+    void emitErrors(List<AsyncError> errors) {
+      for (var error in errors) {
+        _sink.writeln('${error.error}');
+        _sink.writeln(error.stackTrace.toString().trimRight());
+      }
+    }
+
+    final isLoadSuite = test.suite is LoadSuite;
+    if (isLoadSuite) {
+      // Don't emit any info for 'loadSuite' tests, unless they contain errors.
+      if (errors.isNotEmpty || messages.isNotEmpty) {
+        _sink.writeln('${test.suite.path}:');
+        emitMessages(messages);
+        emitErrors(errors);
+      }
+
+      return;
+    }
+
+    final prefix = failed
+        ? _GithubHelper.failedIcon
+        : skipped
+            ? _GithubHelper.skippedIcon
+            : _GithubHelper.passedIcon;
+    final statusSuffix = failed
+        ? ' (failed)'
+        : skipped
+            ? ' (skipped)'
+            : '';
+
+    var name = test.test.name;
+    if (_printPath && test.suite.path != null) {
+      name = '${test.suite.path}: $name';
+    }
+    _sink.writeln(_helper.startGroup('$prefix $name$statusSuffix'));
+    emitMessages(messages);
+    emitErrors(errors);
+    _sink.writeln(_helper.endGroup);
+  }
+
+  void _onDone(bool? success) {
+    _cancel();
+
+    _sink.writeln();
+
+    final hadFailures = _engine.failed.isNotEmpty;
+    String message =
```

Comment:
Probably doesn't matter in practice but we should use a StringBuffer here

Reply:
Updated to use a StringBuffer